mod/primitives/pkg/encoding/ssz/proof/object_path.go

```diff
@@ -0,0 +1,117 @@
+package proof
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/berachain/beacon-kit/mod/primitives/pkg/encoding/ssz/types"
+	"github.com/berachain/beacon-kit/mod/primitives/pkg/math"
+)
+
+// ObjectPath represents a path to an object in a Merkle tree.
+type ObjectPath string
+
+// Split returns the path split by "/"
+func (p ObjectPath) Split() []string {
+	return strings.Split(string(p), "/")
+}
+
+// GetElemType returns the type of the element of an object of the given type with the given index
+// or member variable name (e.g. 7 for x[7], "foo" for x.foo)
+func GetElemType(element SSZType, indexOrVariableName interface{}) (SSZType, error) {
+	switch t := element.Type(); t {
+	case types.Container:
+		variableName, ok := indexOrVariableName.(string)
+		if !ok {
+			return nil, fmt.Errorf("expected string variable name for Container type, got %T", indexOrVariableName)
+		}
+		return element.(Container[SSZType]).GetFieldByName(variableName), nil
+	case types.Elements:
+		index, ok := indexOrVariableName.(uint64)
+		if !ok {
+			return nil, fmt.Errorf("expected string variable name for Container type, got %T", indexOrVariableName)
+		}
+		return element.(Elements).ElementAtIndex(index), nil
```

Comment:
this is a bit problematic, when traversing (for example) a very large list, gindex calculation would need the _nth_ element to be set when in reality just the element type is needed.